crates/wasmtime/src/runtime/store.rs

```diff
@@ -1952,8 +1956,22 @@ from going any further and to alert what's going on. If this is a security
 issue please reach out to the Wasmtime team via its security policy
 at https://bytecodealliance.org/security.
 "
-        );
-        std::process::abort();
+                );
+                std::process::abort();
+            } else if #[cfg(panic = "abort")] {
+                // Without the standard library but with `panic=abort` then
+                // it's safe to panic as that's known to halt execution. For
+                // now avoid the above error message as well since without
+                // `std` it's probably best to be a bit more size-conscious.
+                let _ = pc;
+                panic!("invalid fault");
+            } else {
+                // Without `std` and with `panic = "unwind"` there's no way to
+                // abort the process portably, so flag a compile time error.
+                compile_error!("either `std` or `panic=abort` must be enabled");
```

Comment:
FWIW, we could add another hook to the system interface header for this, if it is ever requested. Might be worth adding a comment about.

Reply:
Yeah I think that'd work out too, but my hope is that all std-using platforms can funnel into the first case with eprintln and most no_std platforms probably don't want unwinding anyway